src/ResultsActivity/ResultsActivityContainer.js

```diff
@@ -120,14 +123,28 @@ class ResultsActivityContainer extends Component {
     )
   }
 
-  handlePrint() {
+  handlePrint(version) {
     const { raiseAnalyticsEvent } = this.props
 
-    raiseAnalyticsEvent({
-      category: 'Activity list',
-      action: 'Print list',
-    })
-    window.print()
+    if (version === 'redacted') {
+      this.setState({ redactedPrint: true })
+
+      raiseAnalyticsEvent({
+        category: 'Redacted Activity list',
+        action: 'Print list',
+      })
+    }
+
+    if (!version) {
+      this.setState({ redactedPrint: false })
+
+      raiseAnalyticsEvent({
+        category: 'Activity list',
+        action: 'Print list',
+      })
+    }
+
+    setTimeout(() => window.print(), 1000)
```

Comment:
Is this guaranteed to be executed after the offender number has been redacted?

Reply:
No.  I've updated it as follows:

`this.setState({ redactedPrint: true }, () => { window.print() })`

That will wait for the state to update now before kicking the print dialogue in to play.

Reply:
Second parameter in `setState` allows you to pass a callback so might be better do `window.print()` in there rather than using a timeout

Reply:
Technically the offender number, or the hidden version of it, should always be redacted.  There's a test for that now.  It's a case of waiting for the state to update one way or another above so it knows to show it in the print.  So long as we invoke after the state update it will be fine.

Reply:
3 minutes ahead of me on that one!  Your comment to say you'd done it wasn't there when I wrote mine :D. Nice one